src/PrimeTower.cpp

```diff
@@ -40,9 +47,32 @@ PrimeTower::PrimeTower()
     const Scene* scene_pointer = &scene; //Communicate to lambda via pointer to prevent copy.
     std::stable_sort(extruder_order.begin(), extruder_order.end(), [scene_pointer](const unsigned int& extruder_nr_a, const unsigned int& extruder_nr_b) -> bool
     {
+        bool retval;
+
         const Ratio adhesion_a = scene_pointer->extruders[extruder_nr_a].settings.get<Ratio>("material_adhesion_tendency");
         const Ratio adhesion_b = scene_pointer->extruders[extruder_nr_b].settings.get<Ratio>("material_adhesion_tendency");
-        return adhesion_a < adhesion_b;
+
+        EPlatformAdhesion adhesion_type = scene_pointer->current_mesh_group->settings.get<EPlatformAdhesion>("adhesion_type");
+        const unsigned int adhesion_extr = scene_pointer->current_mesh_group->settings.get<ExtruderTrain&>("adhesion_extruder_nr").extruder_nr;
+
+        if (adhesion_type == EPlatformAdhesion::RAFT)
+        {
+            retval = (extruder_nr_a == adhesion_extr);
```

Comment:
So basically the idea of this is that if using the raft, the raft extruder is used as outer shell (would also be good to document that; it's pretty hard to read that from this!)
This presents a problem when using a raft on >2 extruder prints. The rest of the extruders are then no longer sorted by adhesion tendency.

I'd guess that you would need a check like this instead:

```
if(adhesion_type == EPlatformAdhesion::RAFT && extruder_nr_a == adhesion_extr)
{
    return true;
}
```

That way other extruders are still properly ordered among each other in the `else` cases.

I also think this reordering is somewhat out of scope. Is there a reason this needs to change to support multiple-extruders-single-nozzle printers?

Reply:
My idea was to still target printing prime tower's external wall with the same extruder as it's single-extruder base, in case that single-extruder base is still there. After I needed to drop the single-extruder base when the base is at the same level as the first layer of the model (but only in case of shared heater/nozzle), the raft is the only remaining opportunity.
But you are most probably right that my code is messing the print order up, apart the first extruder, when more than 2 extruders are used. This for sure needs to be fixed.
About the need for reordering, as said it is simply an opportunity for ensuring the best adhesion of the prime tower external (solid) wall to its base and should be specific
- to the shared heater/nozzle case
- to the first extruder in the sequence, leaving the order of the others unchanged.

Reply:
My preference would go towards not changing the order at all. If this actually changes the order, it would involve a bunch of print reliability tests for us. In general, it is good style to keep the scope of pull requests small. If we need to improve the order of the prime tower shells, let's do that separately.

Reply:
Please have a look at the updated version.
Take special care of the "adhesion_a > adhesion_b" (that I did not modify compared to my earlier push), that is opposite compared to the "adhesion_a < adhesion_b" that I originally started from and that I still see in the main fork/branch.

Reply:
Yeah I saw that. I was wondering if it was now stored in reverse order and interpreted the other way around somehow, and was planning to test the order thoroughly.

Reply:
Well, I do see a conflict between the comment and the code in the original version:

//Sort from high adhesion to low adhesion.
...
return (adhesion_a < adhesion_b);

Maybe the comment is reversed, but to me it makes sense to build the tower from the outer shell towards the inner and with decreasing adhesion tendency, as the inner parts are anyway confined by the outer ones and therefore more robust aginst poor adhesion (also because apart the outer shell, some or many layers could be fully missing for inner shells and in that case the adhesion of the finally printer inner upper layer is 0 by construction, as there is nothing below to adhere to).
To me it looks that with the code currently in the main branch, the outer shall of the prime tower is printed with the extruder/material with the lowest (worst, I guess) adhesion, which doesn't sound a great idea. Please note that in my setup I found most of the material configured with the same adhesion tendency, so this re-ordering, correct or wrong, was never really performed.